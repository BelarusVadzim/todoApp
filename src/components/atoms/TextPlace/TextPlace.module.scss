@import 'src/variables.scss';

.textPlace {
  font-size: $font-size;
  border: none;
  font-family: $main-font;
  cursor: default;
  min-width: none;
}

.textPlaceLineThrough {
  text-decoration: line-through;

  @extend .textPlace;
}

