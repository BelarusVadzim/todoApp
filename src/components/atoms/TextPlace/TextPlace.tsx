import React from 'react';
import { PropsWithChildren } from 'react';
import style from './TextPlace.module.scss';
import { PropsWithClassName } from 'types';

type TextPlaceProps = PropsWithChildren &
  PropsWithClassName & {
    lineThrough?: boolean;
    draggable?: boolean;
  };

const TextPlace: React.FC<TextPlaceProps> = ({
  children,
  lineThrough,
  className = '',
  draggable,
}) => {
  let combinedClassName = lineThrough
    ? style.textPlaceLineThrough
    : style.textPlace;
  combinedClassName = className
    ? `${combinedClassName} ${className}`
    : combinedClassName;

  return (
    <div
      {...(draggable ? { 'data-draggable': 'true' } : {})}
      className={combinedClassName}
    >
      {children}
    </div>
  );
};

export default TextPlace;
