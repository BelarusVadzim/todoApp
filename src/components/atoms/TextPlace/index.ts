import TextPlace from './TextPlace';

export { TextPlace };
