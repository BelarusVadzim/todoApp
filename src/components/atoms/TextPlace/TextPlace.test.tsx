import TextPlace from './TextPlace';
import { render } from '@testing-library/react';

describe('<TextPlace />', () => {
  it('should render properly', () => {
    const component = render(<TextPlace>LabelChild</TextPlace>);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when draggable lineThrough className specified should render properly', () => {
    const component = render(
      <TextPlace draggable lineThrough className="testClass">
        LabelChild
      </TextPlace>
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
