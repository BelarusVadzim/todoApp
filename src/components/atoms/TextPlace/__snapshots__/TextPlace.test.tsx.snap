// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextPlace /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="textPlace"
  >
    LabelChild
  </div>
</DocumentFragment>
`;

exports[`<TextPlace /> when draggable lineThrough className specified should render properly 1`] = `
<DocumentFragment>
  <div
    class="textPlaceLineThrough testClass"
    data-draggable="true"
  >
    LabelChild
  </div>
</DocumentFragment>
`;
